### PR TITLE
feat: add an empty state for the table

### PR DESCRIPTION
**Pull request by etienneburdet**
## Summary

The goal for this PR is to add a small phrase to indicate that no records were found—but appart from that the table hasn't crashed.

(Internal for Opendatasoft only) Associated Shortcut ticket: [sc-50771](https://app.shortcut.com/opendatasoft/story/50771).

### Changes
I added another alternative render for the body, similar to the loading state.

I could have made it into it's own component, but I don't find the `Body.svelte` too cluttered for now. I'm totally open to it though.

### CODE REVIEW

File: packages/visualizations-react/stories/Table/Table.stories.tsx [hunk 1]
```diff
@@ -76,4 +76,10 @@ export const Unstyled = Template.bind({});
 Unstyled.args = {
     data,
     options: { ...options, unstyled: true },
+};
+
+export const emptyState = Template.bind({});
+emptyState.args = {
+    data: { value: [], loading: false },
+    options: { ...options, emptyStateLabel: 'Neniuj registroj trovitaj...' },
```
Comment (KevinFabre-ods) [inline packages/visualizations-react/stories/Table/Table.stories.tsx:84]: Story should show 'Neniuj registroj trovitaj...'

Prop drilling to `Body.svelte` is missing

**Review by KevinFabre-ods — CHANGES_REQUESTED** [review]
Look goods, we just need to extract it from the options in `TableCard.svelte`

**Comment by KevinFabre-ods** [comment]
@etienneburdet With the latest merges, we have some conflicts.
Could you please resolve them, so that we can do a release ? 

Thanks

File: packages/visualizations-react/stories/Table/Table.stories.tsx [hunk 2]
```diff
@@ -83,8 +83,8 @@ Unstyled.args = {
...
 };
 
-export const Loading = Template.bind({});
-Loading.args = {
-    data: fetchingData,
-    options,
+export const emptyState = Template.bind({});
+emptyState.args = {
+    data: { value: [], loading: false },
+    options: { ...options, emptyStateLabel: 'Neniuj registroj trovitaj...' },
```
Comment (KevinFabre-ods) [inline packages/visualizations-react/stories/Table/Table.stories.tsx:89]: We lost the "loading state" story

**Review by KevinFabre-ods — APPROVED** [review]
LGTM ✅  

We just lost the loading story during the merge

File: packages/visualizations-react/stories/Table/Table.stories.tsx (hunk 2 above)
Comment (etienneburdet) [reply packages/visualizations-react/stories/Table/Table.stories.tsx:95]: should be good 🙈

File: packages/visualizations/src/components/Table/Body.svelte [hunk 3]
```diff
@@ -5,9 +5,17 @@
...
 </script>
 
 <tbody>
+    {#if records.length === 0}
+        <tr>
+            <td colspan={columns.length}>
+                <em>{emptyStateLabel}</em>
+            </td>
+        </tr>
+    {/if}
```
Comment (KevinFabre-ods) [inline packages/visualizations/src/components/Table/Body.svelte:18]: We need to display this UI when `isLoading` is true
image

Thanks to the loading story 😝

**Review by etienneburdet — COMMENTED** [review]
haaaa indeed, pushed a bit too fast.

**Review by KevinFabre-ods — APPROVED** [review]
LGTM

**Pull request merged**